### PR TITLE
update: Small cleanup

### PR DESCRIPTION
**Pull request by mrshu**
* Move dependency management to Poetry

* Cleanup the lid.py and lid_utils.py

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,7 +3,9 @@
 
 `conda activate aya`
 
+Install `poetry` as described in https://python-poetry.org/docs/#installation and run `poetry install`
```
Comment (mrshu) [inline README.md:6]: @AMR-KELEG I am not sure how OK would you be with moving to Poetry (which may make conda not necessary in our case) but I believe there are advantages to it, especially if the CLI script might end up being an installable package one day.

File: install.sh [hunk 2]
```diff
@@ -1,15 +1,5 @@
...
-pip install langdetect
-
-# Install the fasttext python package
-pip install git+https://github.com/facebookresearch/fastText.git
-# This version of numpy is required for fasttext models!
-pip install numpy==1.25.2
-
-pip install black
+#!/bin/bash
+set -euo pipefail
```
Comment (mrshu) [inline install.sh:2]: This is the "unofficial bash strictmode": http://redsymbol.net/articles/unofficial-bash-strict-mode/

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+authors = ["mr.Shu <mr@shu.io>"]
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = "^3.11"
+datasets = "^2.20.0"
+langdetect = "^1.0.9"
+fasttext = {git = "https://github.com/facebookresearch/fastText.git"}
+# This version of numpy is required for fasttext models!
+numpy = "1.25.2"
```
Comment (mrshu) [inline pyproject.toml:14]: They `pyproject.toml` will allow us to preserve things like these

File: poetry.lock
Comment (AMR-KELEG) [inline poetry.lock:1]: Should this file be tracked?
If not, could you please untrack it (maybe adding it to .gitignore as well). Feel free to merge the PR afterward 😄

File: README.md (hunk 1 above)
Comment (AMR-KELEG) [reply README.md:6]: Hi Marek,

I am fine with using `poetry`. I have not used it before, but the instructions you left seem to be enough 👍🏽 
Thanks for the suggestion 😄

File: poetry.lock
Comment (mrshu) [reply poetry.lock:1]: @AMR-KELEG Yes, it does indeed need to be tracked for the installation to be reproducible (e.g. for the dependency versions to be pinned).

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+[tool.poetry]
+name = "aya-mislabeled-samples"
+version = "0.1.0"
+description = ""
+authors = ["mr.Shu <mr@shu.io>"]
```
Comment (mrshu) [inline pyproject.toml]:
```suggestion
authors = ["Amr Keleg <amr.keleg.p@gmail.com>", "mr.Shu <mr@shu.io>"]
```

**Comment by mrshu** [comment]
@AMR-KELEG could you please take another look and then perhaps merge this in?

Thanks!

File: poetry.lock
Comment (AMR-KELEG) [reply poetry.lock:1]: I see, thanks for clarifying!

**Pull request merged**